### DISCUSSION TITLE
Flashing or Lightening Motion Trigger

### DISCUSSION THREAD

**Discussion by ceewanna** [roflcoopter/viseron#675] [General]
Currently my setup will undergo almost full-time recording as a result of flashing or lightening light. 

I have tried lowering the fps but I suppose it's already very low, i.e. 3. For indoor activity I have fps 3 but for outdoor activity which involves vehicle, it requires around 8-12 fps to work in my case. 

I am not so sure if it is possible to set a threshold number which the motion trigger needs to make consecutively to qualify for motion detected.

**Comment by roflcoopter** [comment]
There are a few parameters that you can tweak to get better motion detection for your specific scene.
It varies a bit depending on which motion detector you are using, which one is it?

For instance, for [mog2](https://viseron.netlify.app/components-explorer/components/mog2) you can tweak `area`, `threshold`, `history` and `learning_rate`

**Reply by ceewanna** [reply]
I have tried both algorithms. MOG2 is particularly difficult to comprehend. My understanding is as follows (please correct if I am wrong).

area is the overall area of change (threshold is the change of individual pixel). The smaller area the more sensitive. The smaller threshold the more sensitive too.

history is the number of frames it needs to acquire/absorb the change. The smaller history the less it sees those sudden changes.
for learning rate, so far I leave it to -1.

In background extraction, alpha seems to be the inverse of history for me.

My typical settings are threshold (24-32), alpha 0.01, history (200-400), area (0.01 - 0.03). The lower area requires the higher threshold. My fps is between 3 and 12.

My observation so far is that all the above settings along with masks that I use make the system work fine the whole day. However, every night time when there is rain/thunder storm during nighttime, the system will face with noises. The bright flashing light reflects on the building surface or the under roof for a short time every now and then. I am sure it definitely triggers motion. My idea is if we could have a mechanism in which it would count those noisy signals before concluding that it is a motion it might help in my case.

The counter should accumulate when there was trigger in the previous processing and should clear if there was no trigger in this processing or if it reaches the set limit (which means it qualifies for motion detected). In general, the set limit is 1 as it would treat every trigger as motion detected.

**Reply by roflcoopter** [reply]
Your understanding is correct.
I see what you mean, before motion detection actually had this feature, you could set a number of how many consecutive frames with motion were needed before triggering the actualy motion detection.

I removed it for v2 since it added a lot of complexity, but i see now that it might be a needed feature after all.

Will convert this to an issue and implement it again!

**Reply by ceewanna** [reply]
Another aspect of this is if the recording will start from the first trigger or the last trigger in the consecutive frames that qualified as a motion detected. In my view if the series of triggers are qualified for motion, the recording should start from the first frame. However, I have no idea if the actual implementation is practical/feasible or not. In real time, it seems this would require a buffer to keep those consecutive frames or keep track of the time of the first trigger.

**Reply by roflcoopter** [reply]
Ahye thats a good point! It would be good if the recording started from the first frame.
Viseron does recording by writing shorts segments to disk, so buffering is happening in a way (thats how the lookback feature works)
